### ISSUE TITLE
Input Parameters Not Honored when conf=0 or iou=0 in model.predict

### ISSUE THREAD

**Issue by ido-greenfeld** [Deci-AI/super-gradients#1735]
### 🐛 Describe the bug

When setting `conf=0` or `iou=0` as inputs to the `model.predict` function, the resulting prediction does not adhere to the parameter provided, but instead uses the default parameters.

The issue is specifically observed in the YOLO-NAS implementation, but similar occurrences are noted in other files in the repository as well.

In the file [customizable_detector.py](https://github.com/Deci-AI/super-gradients/blob/6436e4280dc93ef254889683050791d9948457d4/src/super_gradients/training/models/detection_models/customizable_detector.py#L176), the following lines are responsible for this behavior:
```
iou = iou or self._default_nms_iou
conf = conf or self._default_nms_conf
```

As per these lines, if either `conf` or `iou` equals 0, the or statement utilizes the default values instead.

Expected Behavior:
The model.predict function should consistently honor the input parameters, even when `conf=0` or `iou=0` is provided as input.

### Versions

super-gradients==3.5.0

**Comment by shaydeci** [comment]
Thanks @ido-greenfeld , indeed it's a bug.
However, I am not sure allowing iou=0 is a desired behaviour as it would result in only filtering according to the confidence, which can be simply achieved without NMS.
We will introduce a fix soon.

**Comment by ido-greenfeld** [comment]
I agree @shaydeci 
Thanks for the fast response!

**Comment by ido-greenfeld** [comment]
btw, how can I disable the NMS in inference time?

**Comment by BloodAxe** [comment]
Short answer you can't disable it completely but you can set `conf` to 1e-6 and `iou` to 0.9999 which would effectively make NMS step as noop

**Comment by BloodAxe** [comment]
This PR partially address issue raised here

**Comment by BloodAxe** [comment]
Fixed in 3.6 release
